ShelterModule/Controllers/AnnouncementController.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using ShelterModule.Models.Announcements;
+using ShelterModule.Models.Pets;
+using ShelterModule.Models.Shelters;
+using ShelterModule.Services.Interfaces.Announcements;
+using ShelterModule.Services.Interfaces.Pets;
+using ShelterModule.Services.Interfaces.Shelters;
+using static Microsoft.EntityFrameworkCore.DbLoggerCategory;
+
+namespace ShelterModule.Controllers
+
+{
+    [ApiController]
+    [Route("announcements")]
+    public class AnnouncementController : ControllerBase
+    {
+        private readonly IAnnouncementQuery _query;
+        private readonly IAnnouncementCommand _command;
+        private readonly IPetQuery _petQuery;
+        private readonly IPetCommand _petCommand;
+        private readonly IShelterQuery _shelterQuery;
+
+        public AnnouncementController(IAnnouncementQuery query, IAnnouncementCommand command, IPetQuery petQuery, IPetCommand petCommand, IShelterQuery shelterQuery)
+        {
+            _query = query;
+            _command = command;
+            _petQuery = petQuery;
+            _petCommand = petCommand;
+            _shelterQuery = shelterQuery;
+        }
+        /// <summary>
+        ///     Returns an announcement with a given ID
+        /// </summary>
+        /// <param name="id"> ID of the announcement that should be returned </param>
+        /// <returns> Announcement with a given ID </returns>
+        [HttpGet]
+        [Route("{id:guid}")]
+        [ProducesResponseType(typeof(AnnouncementResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(UnauthorizedResponse), StatusCodes.Status401Unauthorized)]
```

Comment:
We can't return this response yet, so we can remove this for now. And we should add an information here that 404 can be returned.